packages/playwright-test/src/worker/workerMain.ts

```diff
@@ -320,17 +320,10 @@ export class WorkerMain extends ProcessRunner {
         return;
       }
 
-      const beforeHooksStep = testInfo._addStep({
-        category: 'hook',
-        title: 'Before Hooks',
-        canHaveChildren: true,
-        forceNoParent: true,
-        wallTime: Date.now(),
-      });
-
+      let params: object | null = null;
       // Note: wrap all preparation steps together, because failure/skip in any of them
```

Comment:
This comment suggests that running `beforeEach` from `test` in separate `runFn()` calls will break "skip inside beforeEach" scenario. I guess we are missing a test for this!